### PR TITLE
test(idtoken): left pad ecdsa sig value to proper length

### PR DESCRIPTION
**Pull request by codyoss**
Sometimes r values are generated with a length of 31. Left padding
is the standard to get them to the proper length. This fixes a flaky
test.

Fixes: #583

### CODE REVIEW

**Comment by codyoss** [comment]
```
go test -count=10000 -failfast -run TestValidateES256  google.golang.org/api/idtoken               
ok      google.golang.org/api/idtoken   3.746s
```

**Comment by bradfitz** [comment]
I think your PR title & commit message should clarify this is a test-only deflake and not a fix of the code itself.

**Comment by bradfitz** [comment]
You going to fix the commit message too? (https://github.com/googleapis/google-api-go-client/pull/585/commits)

File: idtoken/validate_test.go [hunk 1]
```diff
@@ -242,8 +242,11 @@ func createES256JWT(t *testing.T) (string, ecdsa.PublicKey) {
 	if err != nil {
 		t.Fatalf("unable to sign content: %v", err)
 	}
+	rb := r.Bytes()
+	rPadded := make([]byte, es256KeySize)
```
Comment (tbpg) [inline idtoken/validate_test.go:246]: `rPadded` or `lPadded`?

**Comment by codyoss** [comment]
Will fix when squashed. 😄

File: idtoken/validate_test.go (hunk 1 above)
Comment (codyoss) [reply idtoken/validate_test.go]: Done.

**Pull request merged**